app/controllers/api/v1/groups_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Controller for managing groups in the API version 1 namespace.
+class Api::V1::GroupsController < ApplicationController
+  before_action :set_group, only: %i[show update destroy]
+
+  def index
+    @groups = Group.all
+    render json: @groups
+  end
+
+  def show
+    render json: @group
+  end
+
+  def create
+    @group = Group.new(group_params)
```

Comment:
why do you need to use instance variables here? Local variables are more than enough in these cases I guess